apps/tenant-management-webapp-e2e/src/integration/events/events.page.ts

```diff
@@ -127,11 +127,109 @@ class eventsPage {
+    return cy.xpath('//*[@data-testid="stream-form" and @data-state="visible"]//div[@class="modal-title"]');
+  }
+
+  streamModalNameInput() {
+    return cy.xpath('//*[@data-testid="stream-form" and @data-state="visible"]//*[@data-testid="stream-name"]');
+  }
+
+  streamModalDescriptionInput() {
+    return cy.xpath('//*[@data-testid="stream-form" and @data-state="visible"]//*[@data-testid="stream-description"]');
+  }
+
+  streamModalSaveButton() {
+    return cy.xpath('//*[@data-testid="stream-form" and @data-state="visible"]//*[@data-testid="form-save"]');
+  }
+
+  streamModalEventDropdown() {
+    return cy.xpath(
+      '//*[@data-testid="stream-form" and @data-state="visible"]/*[@class="modal"]//*[@data-testid="streamEvents-dropdown"]'
+    );
+  }
+
+  streamModalEventDropdownItem(text) {
+    return cy.xpath(
+      `//*[@data-testid="stream-form" and @data-state="visible"]/*[@class="modal"]//*[@data-testid="streamEvents-dropdown"]/following-sibling::*//*[contains(text(), "${text}")]`
+    );
+  }
+
+  streamModalEventDropdownItems() {
+    return cy.xpath(
+      '//*[@data-testid="stream-form" and @data-state="visible"]/*[@class="modal"]//*[@data-testid="streamEvents-dropdown"]/following-sibling::*//li'
+    );
+  }
+
+  streamModalEventDropdownBackground() {
+    return cy.xpath('//*[@data-testid="streamEvents-dropdown-background"]');
+  }
+
+  streamModalRoleCheckbox(roleLabel) {
+    return cy.xpath(
+      `//*[@data-testid="stream-form"]//tbody/tr/td[@class="role-label" and text()="${roleLabel}"]/following-sibling::td//div[contains(@class, "goa-checkbox-container")]`
+    );
+  }
+
+  streamModalRolesCheckboxes() {
+    return cy.xpath(
+      '//*[@data-testid="stream-form"]//tbody/tr/td[@class="role-label"]/following-sibling::td//div[contains(@class, "goa-checkbox-container")]'
+    );
+  }
+
+  streamNameList() {
+    return cy.get('[data-testid="stream-name"]');
+  }
+
+  streamTableBody() {
+    return cy.xpath('//table[@data-testid="autotest-stream-table"]//tbody');
+  }
+
+  streamDetailsEyeIcon(name) {
+    return cy.xpath(
+      `//*[@data-testid="stream-name" and contains(text(), "${name}")]/following-sibling::td//*[@data-testid="toggle-stream-visibility"]`
+    );
+  }
+
+  streamDetailsEyeOffIcon(name) {
+    return cy.xpath(
+      `//*[@data-testid="stream-name" and contains(text(), "${name}")]/following-sibling::td//*[@data-testid="icon-eye-off"]`
+    );
+  }
+
+  streamDetails() {
+    return cy.get('[data-testid="stream-details"]');
+  }
+
+  streamEditBtn(name) {
+    return cy.xpath(
+      `//*[@data-testid="stream-name" and contains(text(), "${name}")]/following-sibling::td//*[@data-testid="edit-stream"]`
+    );
+  }
+
+  streamDeleteBtn(name) {
+    return cy.xpath(
+      `//*[@data-testid="stream-name" and contains(text(), "${name}")]/following-sibling::td//*[@data-testid="delete-stream"]`
+    );
+  }
+
+  streamModalEventChip(name) {
+    return cy.get(`[content="${name}"]`);
```

Comment:
I tried, can not come up with anything

Reply:
Updated the element to be goa-chip for all chips. In the step to use shadow().get() to find the name under goa-chip elements, so that the name will be searched only within chips, not other UI elements.